synapse/storage/data_stores/main/event_push_actions.py

```diff
@@ -819,24 +845,53 @@ def _rotate_notifs_before_txn(self, txn, rotate_to_stream_ordering):
         # Calculate the new counts that should be upserted into event_push_summary
         sql = """
             SELECT user_id, room_id,
-                coalesce(old.notif_count, 0) + upd.notif_count,
+                coalesce(old.%s, 0) + upd.%s,
                 upd.stream_ordering,
                 old.user_id
             FROM (
-                SELECT user_id, room_id, count(*) as notif_count,
+                SELECT user_id, room_id, count(*) as %s,
```

Comment:
why not just use a name which is appropriate to both notif and unread

```suggestion
                SELECT user_id, room_id, count(*) as cnt,
```

(and change line 848 to be `+ upd.cnt`). then you have fewer things to populate with the interpolation.